src/openqaoa-core/qaoa_components/ansatz_constructor/baseparams.py

```diff
@@ -295,13 +269,164 @@ def __repr__(self):
+            elif each_gatemap.gate_label.n_qubits == 2:
+                pair_qubit_coeffs.append(each_coeff)
+                qubit_pairs.append(type(each_gatemap).__name__)
+
+        return (single_qubit_coeffs, pair_qubit_coeffs, qubit_singles, qubit_pairs)
+    
+    @staticmethod
+    def block_setter(input_object: Union[List[GateMap], Hamiltonian], block_type:Enum):
+        
+        if isinstance(input_object, Hamiltonian):
+            block = HamiltonianMapper.generate_gate_maps(
+                input_object, block_type
+            )
+        elif isinstance(input_object, list):
+            for each_gate in input_object:
+                if isinstance(each_gate, RotationGateMap):
+                    each_gate.gate_label.update_gatelabel(
+                        new_gatemap_type=block_type
+                    )
+                else:
+                    raise TypeError(
+                        f"Input gate is of unsupported type {type(each_gate)}."
+                        "Only RotationGateMaps are supported"
+                    )
+            block = input_object
+        else:
+            raise ValueError(
+                "The input object defining mixer should be a List of RotationGateMaps or type Hamiltonian"
+            )
+        return block
+
+    def reorder_gates_block(self, gates_block, layer_number):
+        """Update the qubits that the gates are acting on after application
+        of SWAPs in the cost layer
+        """
+        for gate in gates_block:
+            
+            if layer_number%2 == 0:
+                mapping = self.final_mapping         
+                gate.qubit_1 = mapping[gate.qubit_1]
+                if gate.gate_label.n_qubits == 2:
+                    gate.qubit_2 = mapping[gate.qubit_2]
+            else:
+                pass
+            
+        return gates_block
+                      
+    @staticmethod
+    def route_gates_list(gates_to_route: List[GateMap], device:DeviceBase, routing_function: Callable) -> List[GateMap]:
+        """
+        Apply qubit routing to the abstract circuit gate list
+        based on device information
+        
+        Parameters
+        ----------
+        gates_to_route: `List[GateMap]`
+            The gates to route
+        device: `DeviceBase`
+            The device on which to run the circuit
+        routing_function: `Callable`
+            The function that accepts as input the device, problem, initial_mapping and
+            outputs the list of gates with swaps
+        """
+        problem_to_solve = [[gate.qubit_1, gate.qubit_2] for gate in gates_to_route
+                            if gate.gate_label.n_qubits == 2]
+        (
+            gate_list_indices,
+            swap_mask,
+            initial_physical_to_logical_mapping,
+            final_mapping
+        ) = routing_function(device, problem_to_solve)
+        
+        print("Initial mapping", initial_physical_to_logical_mapping)
```

Comment:
Why do we print that? Oris that temporally?

Reply:
Yes, its temporary -- will be removed!